### PR TITLE
Move DDS fuzz harness to test-dds-utils

### PR DESCRIPTION
**Pull request by Abe27342**
## Description

Moves the DDS fuzz harness extracted from the interval collection fuzz tests to `@fluid-internal/test-dds-utils`.
This package has a more strict linter configuration, so this also fixes up some linter violations.

[AB#3806](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/3806)

### CODE REVIEW

File: packages/dds/test-dds-utils/src/ddsFuzzHarness.ts [hunk 1]
```diff
@@ -410,7 +425,7 @@ function runTest<TChannelFactory extends IChannelFactory, TOperation extends Bas
 			const channel: ReturnType<typeof model.factory.create> = model.factory.create(
 				dataStoreRuntime,
 				// Use friendly IDs for a reasonable number of DDSes for easier debugging.
-				index < 26 ? String.fromCharCode(index + 65) : random.uuid4(),
+				index < 26 ? String.fromCodePoint(index + 65) : random.uuid4(),
```
Comment (CraigMacomber) [inline packages/dds/test-dds-utils/src/ddsFuzzHarness.ts:428]: wouldn't a uuidv5(index) or just to stringing the index work here? This seems strange, but I think your change is fine.

File: pnpm-lock.yaml [hunk 2]
```diff
@@ -6184,7 +6188,11 @@ importers:
       rimraf: ^4.4.0
       typescript: ~4.5.5
     dependencies:
+      '@fluid-internal/stochastic-test-utils': link:../../test/stochastic-test-utils
```
Comment (CraigMacomber) [inline pnpm-lock.yaml:6191]: THis looks like its left over from testing with linking?

Comment (CraigMacomber) [reply pnpm-lock.yaml:6191]: I just this just means its a workspace dep, so this is what we want.

**Review by CraigMacomber — APPROVED** [review]

File: packages/dds/test-dds-utils/src/ddsFuzzHarness.ts (hunk 1 above)
Comment (Abe27342) [reply packages/dds/test-dds-utils/src/ddsFuzzHarness.ts:428]: the linter specifically complained about using `fromCharCode` and wanted `fromCodePoint`. I'm indifferent for this usage (they return the same thing on the range 65 - 89) so I just satisfied the linter.

using uuidv5 would also work fine, but so would not supplying an ID at all. the intent here is to make failure json more human-readable: it's a lot more easy to reason about fuzz testing 3 clients A, B, and C than it is when the clients have guids. if you have more than 26 clients, you probably aren't doing much by-hand minimization or interpretation so we just switch to guids.

File: pnpm-lock.yaml (hunk 2 above)
Comment (Abe27342) [reply pnpm-lock.yaml:6191]: yeah, that's correct. common-utils is in its own workspace so it doesn't look the same.

**Pull request merged**